### PR TITLE
Re-enable FauxPas, fix warnings

### PR DESCRIPTION
**Pull request by jack-stripe**
r? @bdorfman-stripe

### CODE REVIEW

File: Stripe.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -2353,23 +2353,13 @@
 			isa = XCBuildConfiguration;
 			baseConfigurationReference = 04F39F0D1AEF2AFE005B926E /* StripeiOS Tests-Debug.xcconfig */;
 			buildSettings = {
-				ALWAYS_EMBED_SWIFT_STANDARD_LIBRARIES = YES;
```
Comment (bdorfman-stripe) [inline Stripe.xcodeproj/project.pbxproj]: Did you delete these lines on purpose? It seems like the tests would not compile without them

File: FauxPasConfig/main.fauxpas.json [hunk 2]
```diff
@@ -1,7 +1,7 @@
 // Generated from the Faux Pas GUI
 {
     // Xcode target to check (String)
-    "target": "StripeiOSStatic",
+    "target": "StripeiOS",
```
Comment (bdorfman-stripe) [inline FauxPasConfig/main.fauxpas.json]: Did you change this target intentionally? (from what i remember, you have to actually test both targets as they catch slightly different things? I think our ci script does that)

File: Stripe.xcodeproj/project.pbxproj (hunk 1 above)
Comment (jack-stripe) [inline Stripe.xcodeproj/project.pbxproj]: Yep, these are already present in the .xcconfig, so the setting's unchanged.

File: FauxPasConfig/main.fauxpas.json (hunk 2 above)
Comment (jack-stripe) [inline FauxPasConfig/main.fauxpas.json]: Yeah - I think FP's checker is better for framework targets in general. The only reason we were formerly defaulting to StripeiOSStatic is that it had a minimum iOS target of 7.0, which was useful for APIAvailability stuff. Now that we only support 8.0, I think this makes more sense as the default thing to check. (And the CI script will still check both)

**Comment by bdorfman-stripe** [comment]
LGTM

**Pull request merged**